src/oop_ext/foundation/cached_method.py

```diff
@@ -47,94 +62,104 @@ def GetCacheKey(self, *args, **kwargs):
         if kwargs:
             return AsImmutable(kwargs)
 
-    def _HasResult(self, key):
+        return None
+
+    def _HasResult(self, key: Hashable) -> bool:
```

Comment:
Could `@abstractmethod` be added here? I think that `mypy` will understand it even if the class doesn't have `metaclass=ABCMeta`.

Reply:
Good idea, done!